synthetic_monitoring_shared/ui/status.go

```
@@ -0,0 +1,225 @@
+// parseMetrics retains only scalar samples and labels needed by the explicit
+// diagnostics allowlist below. Arbitrary upstream labels never reach JSON.
+func parseMetrics(reader io.Reader) []sample {
+	var result []sample
+	scanner := bufio.NewScanner(io.LimitReader(reader, 16<<20))
+	scanner.Buffer(make([]byte, 0, 64<<10), 1<<20)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		name, labels, rest := line, map[string]string{}, ""
```

golangci-lint (2.12.2)

[error] ineffectual assignment to name

(ineffassign)